nbclient/util.py

```diff
@@ -4,9 +4,52 @@
 # Distributed under the terms of the Modified BSD License.
 
 import asyncio
+import sys
 import inspect
 
 
+def check_ipython():
+    # original from vaex/asyncio.py
+    IPython = sys.modules.get('IPython')
+    if IPython:
+        IPython_version = tuple(map(int, IPython.__version__.split('.')))
+        if IPython_version < (7, 0, 0):
+            raise RuntimeError(f'You are using IPython {IPython.__version__} while we require'
+                               '7.0.0+, please update IPython')
+
+
+def check_patch_tornado():
+    """If tornado is imported, add the patched asyncio.Future to its tuple of acceptable Futures"""
+    # original from vaex/asyncio.py
+    if 'tornado' in sys.modules:
+        import tornado.concurrent
+        if asyncio.Future not in tornado.concurrent.FUTURES:
+            tornado.concurrent.FUTURES = tornado.concurrent.FUTURES + (asyncio.Future, )
+
+
+def just_run(coro):
+    """Make the coroutine run, even if there is an event loop running (using nest_asyncio)"""
+    # original from vaex/asyncio.py
+    try:
+        loop = asyncio.get_event_loop()
```

Comment:
`asyncio.get_event_loop()` will create an event loop if no one already exists, so can we say we had a loop?

Reply:
I thought it did not? But please double check

https://docs.python.org/3.6/library/asyncio-eventloops.html#asyncio.AbstractEventLoopPolicy.get_event_loop

`Raises an exception in case no event loop has been set`

Reply:
That's python3.6, now we have: https://docs.python.org/3/library/asyncio-eventloop.html

Reply:
so... whow :) That's kind of a breaking change isn't it... any idea what the best behavior should be?

Reply:
Actually I'm not sure it has changed. But in any case, we could try and run the coroutine and if we catch an `event loop already running` exception, then we had a loop.

Reply:
It depends on the policy it seems: https://docs.python.org/3.6/library/asyncio-eventloops.html#asyncio.AbstractEventLoopPolicy.get_event_loop

Reply:
Let's say we support only the default policy.

Reply:
I think you are right that the only way to detect a running loop in py36 is to just try :(